cmd/swctl/app/dialog.go

```diff
@@ -0,0 +1,135 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+	"github.com/manifoldco/promptui"
+)
+
+type PromptContent struct {
+	Label    string
+	MsgError string
+	// attribute items is needed only for select and multi-select
+	Items []string
+}
+
+func (pc *PromptContent) PromptGetInput() []string {
+	validate := func(input string) error {
+		if len(input) <= 0 {
+			return errors.New(pc.MsgError)
+		}
+		return nil
+	}
+	templates := &promptui.PromptTemplates{
+		Prompt:  "{{ . }}",
+		Valid:   "{{ . | green}}",
+		Invalid: "{{ . | red}}",
+		Success: "{{ . | bold}}",
+	}
+
+	prompt := promptui.Prompt{
+		Label:     pc.Label,
+		Validate:  validate,
+		Templates: templates,
+	}
+
+	result, err := prompt.Run()
+	if err != nil {
+		fmt.Printf("Prompt failed %v\n", err)
+	}
+	fmt.Printf("Input: %s\n", result)
+	return []string{result}
+}
+
+func (pc *PromptContent) PromptGetSelect() []string {
+	index := -1
+	var items []string
+	var result string
+	var err error
+
+	for index < 0 {
+		prompt := promptui.Select{
+			Label: pc.Label,
+			Items: pc.Items,
+		}
+		index, result, err = prompt.Run()
+		if index == -1 {
+			items = append(items, result)
+		}
+
+	}
+	if err != nil {
+		fmt.Println("Prompt failed %v\n", err)
+	}
+	fmt.Printf("Input: %s\n", result)
+	return []string{result}
+}
+
+type localItem struct {
+	Id         string
+	IsSelected bool
+}
+
+func (pc *PromptContent) PromptGetMultiSelect() []string {
+	fmt.Print(pc.Label)
+	const DoneId = "done"
+	var usedStrings = pc.Items
+
+	var items []*localItem
+	for _, name := range usedStrings {
+		items = append(items, &localItem{Id: name})
+	}
+	items = append(items, &localItem{Id: DoneId})
+	selectedItems := localPromptGetMultiSelect(0, items, pc.Label)
+	var stringerizedSelectedItems []string
+	for _, buf := range selectedItems {
```

Comment:
What is `buf`? Does it have some special logical meaning to the selectedItems? When not, then there are more generic names to use,. Preferably singular to list variable name (`selectedItem`) or some short name (`item`, `si`) or some other thing that doesn't bring confusion (i.e. `i` can confuse as it is used for indexing but that is not used here due to `_` usage).